### PR TITLE
Allow use of StringMap for Context

### LINKED ISSUE #8: Use StringMap for Context
Is there any reason why is context forced to be Dynamic? My backend uses StringMap and I am getting into serious problems making it work.  Plus I think that somthing like StringMap should be faster than using reflection. 
Was there any design decision why to use Dynamic?

### PR DESCRIPTION
**Pull request by Tezar**
resolves #8

### CODE REVIEW

File: src/mustache/Context.hx [hunk 1]
```diff
@@ -80,7 +80,13 @@ private class ContextImpl {
     }
 
     static function getField(object:Dynamic, name:String):haxe.ds.Option<Dynamic> {
-        var value = Reflect.field(object, name);
+        var value;
```
Comment (nadako) [inline src/mustache/Context.hx]: No need to declare this var, just move the whole `var value = Reflect.field(object, name)` expression after the check

File: src/mustache/Context.hx [hunk 2]
```diff
@@ -80,7 +80,13 @@ private class ContextImpl {
...
+        if (Std.is(object, haxe.ds.StringMap)) {
```
Comment (nadako) [inline src/mustache/Context.hx]: `Std.downcast` will do the type check, so there's no need for `Std.is` here, simply do
```haxe
var map = Std.downcast(object, haxe.ds.StringMap);
if (map != null) {
    return map.exists(name) ? Some(map.get(name)) : None;
}
```

**Review by nadako — CHANGES_REQUESTED** [review]

**Comment by Tezar** [comment]
Looking at travis, should I bump haxe version to 4.0.1?

**Comment by nadako** [comment]
Hmm, maybe better do `#if haxe4 Std.downcast #else Std.instance #end(object, haxe.ds.StringMap)` or something for now, because it's the only thing that requires Haxe 4, right?

**Comment by Tezar** [comment]
Reread why downcast was introduced and seems ok to use instance on hx3. Patched as suggested.

Build still fails but that's due to problem with buddy tools.

**Comment by nadako** [comment]
Oh well, still merged

**Comment by Tezar** [comment]
Could I bother you some more with new release to haxe lib? :)

Thanks

**Pull request merged**